### ISSUE TITLE
could taskReader persistAckLevel check last persist ackLevel avoid persist duplicate ackLevel

### ISSUE THREAD

**Issue by zedongh** [temporalio/temporal#4692]
**Is your feature request related to a problem? Please describe.**
Some workflow worker or actvitiy worker just startup wait for task coming but maybe hold for long time. Here are may no task coming for a long while. Persist acklevel cause effortless database request per updateAckTimer.

**Describe the solution you'd like**
`taskReader` store last persist ack level and do `persistAckLevel` only when current ack level not match last persist ack level.

**Describe alternatives you've considered**
No

**Additional context**
No

**Comment by dnr** [comment]
That's correct. Actually there are more optimizations that could be done in that area, e.g. not trying to read tasks from the database periodically either. I've been looking into this area recently so there may be some changes here soon.
